runtime/compiler/control/CompilationThread.cpp

```diff
@@ -13181,15 +13181,17 @@ bool TR::CompilationInfo::canProcessJProfilingRequest()
 bool
 TR::CompilationInfo::canRelocateMethod(TR::Compilation *comp)
    {
+#if defined(J9VM_OPT_JITSERVER)
+   // A deserialized JITServer AOT cache method won't be stored in the local SCC, so it must be
+   // relocated immediately always
+   if (comp->isDeserializedAOTMethod())
+      return true;
+#endif /* defined(J9VM_OPT_JITSERVER) */
+
    // Delay relocation by default, unless this option is enabled
    if (!comp->getOption(TR_DisableDelayRelocationForAOTCompilations))
       return false;
 
-#if defined(J9VM_OPT_JITSERVER)
-   if (comp->isDeserializedAOTMethod() && comp->getPersistentInfo()->getJITServerAOTCacheDelayMethodRelocation())
```

Comment:
I've removed it.

Reply:
Just to update this - I haven't removed the option in this final version of the PR, because we still have the option of running with the old implementation.